### PR TITLE
Enable culture analyzer and fix issues

### PR DESCRIPTION
**Pull request by Evangelink**
I have tried to group changes per project, set of projects.

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs [hunk 1]
```diff
@@ -478,7 +478,7 @@ private void LoadAndInitialize(DataCollectorSettings dataCollectorSettings, stri
 
             if (!IsUriValid(dataCollectorUri) && !TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri))
             {
-                LogWarning(string.Format(CultureInfo.CurrentUICulture, Resources.Resources.UnableToFetchUriString, dataCollectorSettings.FriendlyName));
+                LogWarning(string.Format(CultureInfo.CurrentCulture, Resources.Resources.UnableToFetchUriString, dataCollectorSettings.FriendlyName));
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs:481]: What is the reason for changing this from ui culture to current culture?

**Comment by nohwnd** [comment]
Imho in testhost we are setting current_UI culuture for the process and so I guess not using currentUIculture for formatting changes how messages are written to the screen. Did you do any tests to make sure that is not the case?

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs (hunk 1 above)
Comment (Evangelink) [reply src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs:481]: The analyzer is complaining about it :) 

[Rule documentation](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1304) says:

> [CultureInfo.CurrentUICulture](https://docs.microsoft.com/dotnet/api/system.globalization.cultureinfo.currentuiculture) is used only to retrieve localized resources by using an instance of the [System.Resources.ResourceManager](https://docs.microsoft.com/dotnet/api/system.resources.resourcemanager) class.

**Comment by Evangelink** [comment]
@nohwnd I haven't done any special test, I was expecting our test suite to catch translation issues. I will try to see if I can run something manually to confirm things are ok.

**Comment by nohwnd** [comment]
I doubt that will happen, all the tests are in english, and all the computers are running in english.

**Comment by Evangelink** [comment]
@nohwnd I am too used to roslyn where one CI is running in foreign language to ensure translation are tested.

**Comment by Evangelink** [comment]
I have run a couple of manual tests using French culture and I haven't seen any problem. Do you know any special test we can run?

**Comment by nohwnd** [comment]
> I have run a couple of manual tests using French culture and I haven't seen any problem. Do you know any special test we can run?

Does that mean you are able to run on non-english system, and revert the output back to english usnig that env variable and vice versa? That is imho the purpose of that env variable.

**Review by nohwnd — APPROVED** [review]

**Pull request merged**